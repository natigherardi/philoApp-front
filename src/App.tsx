import MainContainer from "./AppStyled";
import { Routes, Route, Navigate } from "react-router-dom";
import RegisterPage from "./pages/RegisterPage/RegisterPage";

function App() {
  return (
    <MainContainer>
      <Routes>
        <Route path="/" element={<Navigate to={"/register"} />} />
        <Route path="/register" element={<RegisterPage />} />
      </Routes>
    </MainContainer>
  );
}

export default App;
