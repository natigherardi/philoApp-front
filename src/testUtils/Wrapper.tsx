import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import WrapperProps from "../types/Wrapper";
import mockStore from "./mockStore";

const Wrapper = ({ children }: WrapperProps) => {
  return (
    <BrowserRouter>
      <Provider store={mockStore}>{children}</Provider>
    </BrowserRouter>
  );
};

export default Wrapper;
