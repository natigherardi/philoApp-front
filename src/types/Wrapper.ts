interface WrapperProps {
  children: JSX.Element | JSX.Element[];
}

export default WrapperProps;
