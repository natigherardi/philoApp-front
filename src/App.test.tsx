import TestRenderer from "react-test-renderer";
import App from "./App";
import Wrapper from "./testUtils/Wrapper";

describe("Given the app component", () => {
  describe("When rendered", () => {
    test("Then it should always match this snapshot", () => {
      const expectedApp = TestRenderer.create(
        <Wrapper>
          <App />
        </Wrapper>
      );

      expect(expectedApp).toMatchSnapshot();
    });
  });
});
