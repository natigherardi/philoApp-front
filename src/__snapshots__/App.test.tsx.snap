// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the app component When rendered Then it should always match this snapshot 1`] = `
<div
  className="sc-bczRLJ jYHDTx"
>
  <div
    className="sc-crXcEl kEztYC"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-x modal__close-icon"
      data-icon="x"
      data-prefix="fas"
      data-testid="close-icon"
      focusable="false"
      onClick={[Function]}
      role="img"
      style={Object {}}
      viewBox="0 0 384 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M376.6 84.5c11.3-13.6 9.5-33.8-4.1-45.1s-33.8-9.5-45.1 4.1L192 206 56.6 43.5C45.3 29.9 25.1 28.1 11.5 39.4S-3.9 70.9 7.4 84.5L150.3 256 7.4 427.5c-11.3 13.6-9.5 33.8 4.1 45.1s33.8 9.5 45.1-4.1L192 306 327.4 468.5c11.3 13.6 31.5 15.4 45.1 4.1s15.4-31.5 4.1-45.1L233.7 256 376.6 84.5z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="modal__title"
    >
      Success âœ“
    </span>
    <span
      className="modal__description"
    >
      no message
    </span>
  </div>
</div>
`;
